src/inmanta/data/sqlalchemy.py

```diff
@@ -45,6 +45,66 @@ class Base(DeclarativeBase):
     pass
 
 
+class Module(Base):
+    __tablename__ = "module"
+
+    __table_args__ = (
+        PrimaryKeyConstraint("name", "version", "environment", name="module_pkey"),
+        ForeignKeyConstraint(["environment"], ["environment.id"], ondelete="CASCADE", name="code_environment_fkey"),
+    )
+
+    name: Mapped[str] = mapped_column(String)
+    version: Mapped[str] = mapped_column(String)
+    environment: Mapped[uuid.UUID] = mapped_column(UUID)
+    requirements: Mapped[list[str]] = mapped_column(ARRAY(String()))
+
+
+class FilesInModule(Base):
+    __tablename__ = "files_in_module"
+    __table_args__ = (
+        ForeignKeyConstraint(
+            ["module_name", "module_version", "environment"],
+            ["module.name", "module.version", "module.environment"],
+            ondelete="CASCADE",
+        ),
+        ForeignKeyConstraint(["environment"], ["environment.id"], ondelete="CASCADE", name="files_in_module_environment_fkey"),
+        ForeignKeyConstraint(
+            ["file_content_hash"], ["file.content_hash"], ondelete="RESTRICT", name="files_in_module_file_content_hash_fkey"
+        ),  # todo add test for restrict behaviour on delete
+        UniqueConstraint("module_name", "module_version", "environment", "file_path"),
+    )
+    __mapper_args__ = {"primary_key": ["module_name", "module_version", "environment", "file_path"]}
+
+    module_name: Mapped[str] = mapped_column(String)
+    module_version: Mapped[str] = mapped_column(String)
+    environment: Mapped[uuid.UUID] = mapped_column(UUID)
+    file_content_hash: Mapped[str] = mapped_column(String)
+    file_path: Mapped[str] = mapped_column(String)
```

Comment:
1. is this the path where it comes from or where it should go
2. path relative to what
3. Do we need/want this? Would it not be more convenient to use just module name and if it a py or a pyc file?

Reply:
- Updated `file_path` -> `python_module_name`
- Added `is_byte_code`